aya/src/programs/tc.rs

```diff
@@ -124,19 +127,28 @@ impl SchedClassifier {
         interface: &str,
```

Comment:
I think we can probably make the API a bit nicer. Now most people will be calling it like this (as per our own doc example):

```
let prog: &mut SchedClassifier = bpf.program_mut("redirect_ingress").unwrap().try_into()?;
prog.load()?;
prog.attach("eth0", TcAttachType::Ingress, 0, 0)?;
```

That "0, 0" isn't great. Maybe we should have:

```
prog.attach("eth0", TcAttachType::Ingress);
```

and for people that want more control

```
prog.attach_with_options("eth0", TcAttachType::Ingress, priority, handle)
```

Suggestions for alternatives to the `attach_with_options` name welcome

Reply:
I don't like the 0's either.  We could make this more readable by using constants.  For example:
```
prog.attach("eth0", TcAttachType::Ingress, tc::AUTO_PRI, tc::AUTO_HNDL)?;
```

I think whether we should have two versions depends on whether most users would want to explicitly use the options.  If they don't then it makes sense to have the "simple" and "advanced" versions.  

I suspect that many would want to set the priority, but most would not need to choose the handle. What do you think?

Reply:
Having the constants like that wouldn't be very rusty I think. If we wanted to go that route, we could have

```
#[derive(Default)]
struct TcAttachPriotity(u32);

#[derive(Default)]
struct TcAttachHandle(u32);
```

And then you would call it with `attach("eth0", TcAttachType::Ingress, TcAttachPriority::default(), TcAttachHandle::default())`, or `attach("eth0", TcAttachType::Ingress, TcAttachPriotity(123), TcAttachHandle(345))`. It still seems a bit much to me.

I have personally never passed priority and handle, so to me two methods seems the best approach. But I've never had to integrate with any other TC programs, I genuinely don't know how common that is.

Reply:
If we went the `TcAttachPriotity` and `TcAttachHandle` way we could actually have

```
fn attach<P: Into<TcAttachPriotity>, H: Into<TcAttachHandle>>(name: &str, attach_type: TcAttachType, priority: P, handle: H) -> ... {
   let priority = priority.into();
   let handle = handle.into();
  ...
}
```

and then call it as 

```
attach("eth0", TcAttachType::Ingress, TcAttachPriority::default(), TcAttachHandle::default())
```

or 

```
attach("eth0", TcAttachType::Ingress, 123, 456)
```

which is more characters but better than `0, 0`

Reply:
I think this is now my preference :)

Reply:
That works for me.  Thanks.

Reply:
But, I'll hold off until tomorrow to see if anyone else has other ideas :-)